File: tests/golden/fixtures/generator/hbs/js-helper/addons/js/generator/common/helpers/describe.js
// Describe helper: reports type and value in a deterministic string.
// Uses normalize_args and format_object from _utils.js (loaded before helper files).

function describe()
{
    var list = normalize_args(arguments);
    var type;
    var value;

    if (list.length === 0)
    {
        type = 'undefined';
        value = '';
    }
    else if (list.length > 1)
    {
        type = 'array';
        value = list.join(',');
    }
    else
    {
        var v = list[0];
        if (v === null)
        {
            type = 'null';
            value = '';
        }
        else if (Array.isArray(v))
        {
            type = 'array';
            value = v.join(',');
        }
        else
        {
            type = typeof v;
            if (type === 'object')
                value = format_object(v);
            else
                value = String(v);
        }
    }

    return type + ':' + value;
}

File: tests/golden/fixtures/generator/hbs/lua-helper/addons/lua/generator/common/helpers/describe.lua
-- Describe helper: reports type and value in a deterministic string.
-- Uses normalize_args and format_object from _utils.lua.

return function(...)
    local list = normalize_args(...)
    local typ
    local value

    if list.n == 0 then
        typ = "undefined"
        value = ""
    elseif list.n > 1 then
        typ = "array"
        local strs = {}
        for i = 1, list.n do
            strs[i] = tostring(list[i])
        end
        value = table.concat(strs, ",")
    else
        local v = list[1]
        if v == nil then
            typ = "null"
            value = ""
        else
            typ = type(v)
            if typ == "table" then
                value = format_object(v)
            else
                value = tostring(v)
            end
        end
    end

    return typ .. ":" .. value
end
